intake/catalog/base.py

```diff
@@ -237,17 +374,35 @@ def _load(self):
         if self._source_id is not None:
             headers['source_id'] = self._source_id
         http_args['headers'] = headers
-        response = requests.get(self.info_url, **http_args)
+        return http_args
+
+    def _load(self):
+        """Fetch metadata from remote. Entries are fetched lazily."""
+        # This will not immediately fetch any sources (entries). It will lazily
+        # fetch sources from the server in paginated blocks when this Catalog
+        # is iterated over. It will fetch specific sources when they are
+        # accessed in this Catalog via __getitem__.
+
+        # Just fetch the metadata.
+        params = {'page[number]': 1, 'page[size]': 0}
+        response = requests.get(self.info_url, params=params,
+                                **self._get_http_args())
         if response.status_code != 200:
             raise Exception('%s: status code %d' % (response.url,
                                                     response.status_code))
         info = msgpack.unpackb(response.content, encoding='utf-8')
         self.metadata = info['metadata']
-
-        self._entries = {s['name']: RemoteCatalogEntry(
-            url=self.source_url,
-            getenv=self.getenv,
-            getshell=self.getshell,
-            auth=self.auth,
-            http_args=self.http_args, **s)
-            for s in info['sources']}
+        self._entries.reset()
+        # If the server respects the pagination parameters, info['sources']
+        # should be empty. But if we have an old server, it will contain all
+        # the entries and we should cache them now.
+        if info['sources']:
+            self._server_can_paginate = False
+            self._entries._page_cache.update(
```

Comment:
why not just set the dictionary as before in this case?

Reply:
Updating the existing cache feels safer than blowing away the whole object and replacing it. But I can appreciate the simplicity of that approach, and could be persuaded.